modules/gateway/peers_test.go

```diff
@@ -236,38 +234,46 @@ func TestListen(t *testing.T) {
 		t.Fatal("gateway should have given ack")
 	}
 
-	if build.VersionCmp(build.Version, sessionUpgradeVersion) >= 0 {
-		// generate fake ID
-		var gID gatewayID
-		fastrand.Read(gID[:])
-		wantConnect := true
-		// continue handshake protocol, this time exchanging the session header
-		err = connectSessionHandshake(conn, gID, wantConnect)
-		if err != nil {
-			t.Fatal(err)
-		}
+	header.NetAddress = modules.NetAddress(conn.LocalAddr().String())
+	err = writeSessionHeader(conn, header)
+	if err != nil {
+		t.Fatal(err)
 	}
-
-	err = connectPortHandshake(conn, addr.Port())
+	_, err = readSessionHeader(conn, header)
 	if err != nil {
 		t.Fatal(err)
 	}
 
 	// g should add the peer
-	var ok bool
-	for !ok {
+	err = build.Retry(10, 20*time.Millisecond, func() error {
```

Comment:
you are probably going to want more time than this, Travis can sleep a thread for as long as 5 seconds iirc